### DISCUSSION TITLE
Request hangs when selfHandleResponse: true

### DISCUSSION THREAD

**Discussion by CanLikeTo** [chimurai/http-proxy-middleware#810] [Q&A] [answered]
Hi, I'm trying to set up a simple dev proxy where the auth endpoints are proxied to another service. Basically to avoid CORS issues. In order to make this work, I have to rewrite the host part of the `set-cookie` header. I've been following the [response-interceptor recipe](https://github.com/chimurai/http-proxy-middleware/blob/master/recipes/response-interceptor.md#readme) and basically my request hangs and eventually times out as soon as I set `selfHandleResponse: true`.

This is my setup:

```js
const app = express();

app.use(
  '/api/auth',
    createProxyMiddleware({
      target: 'https://playground.projects.oryapis.com',
      changeOrigin: true,
      pathRewrite: {
        '/api/auth': '/'
      },
      logger: console,
      plugins: [simpleRequestLogger],
      secure: false,
      selfHandleResponse: true,
      on :{
        proxyRes: responseInterceptor(async (responseBuffer, proxyRes, req, res) => {

          return responseBuffer;
        })
      }
    })
);

app.use(
  '/',
    createProxyMiddleware({
      target: 'http://localhost:3000',
      logger: console,
    })
);
```

My versions are as follows:
```
node: v16.14.2
http-proxy-middleware: 2.0.6
express: 4.17
```
I'm also developing on Github Codespaces.

**Comment by chimurai** [comment, marked as answer]
Dupe of https://github.com/chimurai/http-proxy-middleware/issues/807#issuecomment-1193359576

**Reply by CanLikeTo** [reply]
Thank you very much!
